lms/services/assignment.py

```diff
@@ -289,78 +288,27 @@ def get_assignments(  # noqa: PLR0913
             )
 
         if course_ids:
-            deduplicated_course_assignments = (
-                self._deduplicated_course_assigments_query(course_ids).subquery()
-            )
-
-            query = query.where(
-                # Get only assignment from the candidates above
-                Assignment.id == deduplicated_course_assignments.c.assignment_id,
-                deduplicated_course_assignments.c.grouping_id.in_(course_ids),
-            )
+            query = query.where(Assignment.course_id.in_(course_ids))
 
         return query.order_by(Assignment.title, Assignment.id).distinct()
 
-    def _deduplicated_course_assigments_query(self, course_ids: list[int]):
-        # Get all assignment IDs we recorded from this course
-        raw_course_assignments = select(AssignmentGrouping.assignment_id).where(
-            AssignmentGrouping.grouping_id.in_(course_ids)
-        )
-
-        # Get a list of deduplicated assignments based on raw_course_assignments,
-        # this will contain assignments that belong (now) to other courses
-        return (
-            select(AssignmentGrouping.assignment_id, AssignmentGrouping.grouping_id)
-            .distinct(AssignmentGrouping.assignment_id)
-            .join(Grouping)
-            .where(
-                # Only look at courses, otherwise courses and sections will deduplicate each other
-                Grouping.type == "course",
-                # Use the previous query to look only at the potential candidates
-                AssignmentGrouping.assignment_id.in_(raw_course_assignments),
-            )
-            # Deduplicate them based on the updated column, take the last one (together with the distinct clause)
-            .order_by(
-                AssignmentGrouping.assignment_id, AssignmentGrouping.updated.desc()
-            )
-        )
-
-    def get_courses_assignments_count(
-        self, course_ids: list[int], **kwargs
-    ) -> dict[int, int]:
+    def get_courses_assignments_count(self, **kwargs) -> dict[int, int]:
         """Get the number of assignments a given list of courses has."""
-
-        assignments_query = (
+        query = (
             # Query assignments
             self.get_assignments(**kwargs)
-            # Only select their IDs
-            .with_only_columns(Assignment.id)
-            # Remove any sorting options, we are going to avoid having to worry about sorted columns
-            .order_by(None)
-        )
-
-        # We didn't pass course_ids to get_assigments because we need to deduplicate when we count, not before
-        deduplicated_course_assignments = self._deduplicated_course_assigments_query(
-            course_ids
-        ).subquery()
-
-        counts_query = (
-            select(
-                AssignmentGrouping.grouping_id,
-                func.count(AssignmentGrouping.assignment_id),
+            # Change the selected columns
```

Comment:
This is much simpler now, just write the query we need instead of having to shoehorn the deduplicating step.